train.py

```diff
@@ -135,23 +140,22 @@ def train_epoch(
             for p in model.parameters():
                 param_norm = p.grad.detach().data.norm(2)
                 total_norm += param_norm.item() ** 2
-            b += 1 if total_norm ** 0.5 < S else 0
+            b += 1 if total_norm ** 0.5 < S_e else 0
             b += torch.randn(1) * sigma_b
```

Comment:
I think we can also move this line to line 155 and 158 so that the noise is only added for each batch